util.py

```diff
@@ -409,7 +410,7 @@ def __init__(self, duration, max_mem_mb = None):
 
         #Set the duration
         self._duration = int(duration)
-        signal.signal(signal.SIGALRM, Watchdog.self_destruct)
+        self._timer = threading.Timer(self._duration, Watchdog.self_destruct)
```

Comment:
I might be confused, but if the collector goes crazy during the very first run, the timer is not activated because `.reset()` is only called at the end of the loop and the timer hasn't been started yet.